File: solutions/fsharp/acronym/1/Acronym.fs
module Acronym

let abbreviate (phrase: string): string =
    phrase.Split([|' '; '\n'; '\t'; '\v'; '\f'; '\r'; '-'; '_'|], System.StringSplitOptions.RemoveEmptyEntries)
    |> Array.map (fun (x:string) -> x.[0])
    |> fun x -> (new string(x)).ToUpper()

File: solutions/haskell/acronym/1/src/Acronym.hs
module Acronym (abbreviate) where

import Data.Char (isAlpha, isLower, isSpace, isUpper, toUpper)

m :: (Char, Char) -> String
m (a, b) | not (isAlpha a || a == '\'') && isAlpha b = [toUpper b]
m (a, b) | isLower a && isUpper b = [toUpper b]
m _ = []

abbreviate :: String -> String
abbreviate s = concatMap m $ zip (" " <> s) s
